pkg/command/service/generate.go

```diff
@@ -69,50 +78,31 @@ kperf service generate —n 500 —-interval 20 —-batch 20 --min-scale 0 --max
 				}
 				if start > 0 && end > 0 && start <= end {
 					for i := start; i <= end; i++ {
-						namespaceRangeMap[fmt.Sprintf("%s-%d", generateArgs.namespacePrefix, i)] = true
+						nsNameList = append(nsNameList, fmt.Sprintf("%s-%d", generateArgs.namespacePrefix, i))
 					}
 				} else {
 					return errors.New("failed to parse namespace range")
 				}
+			} else if generateArgs.namespace != "" {
+				nsNameList = append(nsNameList, generateArgs.namespace)
 			}
-			ksvcClient, err := p.NewServingClient()
-			if err != nil {
-				return err
+
+			if len(nsNameList) == 0 {
+				return fmt.Errorf("no namespace found with prefix %s", generateArgs.namespacePrefix)
 			}
-			nsNameList := []string{}
-			if generateArgs.namespace != "" {
-				nss, err := p.ClientSet.CoreV1().Namespaces().Get(context.TODO(), generateArgs.namespace, metav1.GetOptions{})
-				if err != nil {
-					return err
-				}
-				nsNameList = append(nsNameList, nss.Name)
-			} else if generateArgs.namespacePrefix != "" {
-				nsList, err := p.ClientSet.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
-				if err != nil {
-					return err
-				}
-				if len(nsList.Items) == 0 {
-					return fmt.Errorf("no namespace found with prefix %s", generateArgs.namespacePrefix)
-				}
-				if len(namespaceRangeMap) >= 0 {
-					for i := 0; i < len(nsList.Items); i++ {
-						if _, exists := namespaceRangeMap[nsList.Items[i].Name]; exists {
-							nsNameList = append(nsNameList, nsList.Items[i].Name)
-						}
-					}
-				} else {
-					for i := 0; i < len(nsList.Items); i++ {
-						if strings.HasPrefix(nsList.Items[i].Name, generateArgs.namespacePrefix) {
-							nsNameList = append(nsNameList, nsList.Items[i].Name)
-						}
-					}
+			// Check if namespace exists, in NOT, return error
+			for _, ns := range nsNameList {
```

Comment:
If --namespace-prefix "test" --namespace-range is "1,3", but there are only test-1 and test-3, the old implementation can work but the new one can not.

Reply:
And what about the same code snippet in clean?

Reply:
@qibobo I'm thinking this is work as design, as described in the README, users should prepare the namespaces. For e.g., our test case is to distribute the service in 10 namespaces, test-1 to test-10, if test-2 to test-9 not working, the service will be disctributed to 2 namespaces, this will impact the test result due to autoscaler behavior, WDYT?

Reply:
Make sense.

Reply:
For cleanup, i think the original works fine, if no namespace found, we can just skip